File: AN_2/SEM_2/MPP/java/lab3/src/main/java/ro/mpp2024/repository/RepositoryInterface.java
package ro.mpp2024.repository;

import java.util.List;

public interface RepositoryInterface<ID,T> {
    void adauga(T entity);
    void sterge(ID id);
    void modifica(T entity);
    T findById(ID id);
    List<T> findAll();

}

File: AN_2/SEM_2/MPP/c#/lab2_1/repository/RepositoryInterface.cs
﻿namespace lab2_1.repository;

public interface RepositoryInterface
{
    public interface IRepository<T>
    {
        void adauga(T entity); 
        void sterge(int id);
        void modifica(T entity);
        T findById(int id);
        List<T> findAll();


    }
}